torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1937,52 +1937,50 @@ std::vector<Val*> Index::getNonGlobalProducerStridedIndices(
   return strided_inds;
 }
 
-std::vector<Val*> Index::getLinearIndex(
-    TensorView* consumer_tv,
-    const std::vector<kir::ForLoop*>& loops) {
+std::vector<Val*> Index::getRootIndicesWithContiguity(
+    TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops,
+    bool contiguity) {
   // Use domain guard to ignore the contiguity of
   //  consumer tv.
-  TensorDomain* consumer_tv_no_contiguity_domain = nullptr;
-  auto contiguity_vector =
-      std::vector<bool>(consumer_tv->getMaybeRFactorDomain().size(), true);
-  if (consumer_tv->hasRFactor()) {
-    consumer_tv_no_contiguity_domain = IrBuilder::create<TensorDomain>(
-        consumer_tv->getRootDomain(),
-        consumer_tv->getRFactorDomain(),
-        consumer_tv->domain()->domain(),
+  TensorDomain* no_contiguity_domain = nullptr;
```

Comment:
nit: probably shouldn't be called `no_contiguity_domain` anymore?

We used an option for rng which is all true for contiguity. It looks like we need another all false here. Is that correct?

Reply:
> We used an option for rng which is all true for contiguity. It looks like we need another all false here. Is that correct?

Yes, correct.

> probably shouldn't be called `no_contiguity_domain` anymore?

I agree, will change it to `domain_with_specified_contiguity`, if this sounds good to you.

Reply:
fixed